### ISSUE TITLE
Feature Request: Definition for multiple `BSP` connections in one json file

### ISSUE THREAD

**Issue by Gedochao** [build-server-protocol/build-server-protocol#337]
<img width="396" alt="intellij_bsp_build_scripts_found" src="https://user-images.githubusercontent.com/18601388/177160345-dd8b776a-038c-4b2c-b150-a518f19a463b.png">

`IDEA IntelliJ` currently allows for multiple `BSP` modules to be imported in a single project. As visible on the screenshot above, the presence of multiple definitions in different sub-directories is even detected by the IDE (although the actual modules have to be added by hand).

I'm thinking if `BSP` could define a centralised way of defining multiple connections.

The current way of defining multiple `BSP` connections:

```text
.
├── app1
│   ├── .bsp
│   │   └── scala-cli.json
│   └──  ...
├── app2
│   ├── .bsp
│   │   └── scala-cli.json
│   └──  ...
└── scripts
    ├── .bsp
    │   └── scala-cli.json
    └──  ...
```

Instead, I believe we could have a single `.bsp` definition covering a couple modules (with one probably set as the default one):

```text
.
├── .bsp
│   └── scala-cli.json
├── app1
│   └──  ...
├── app2
│   └──  ...
└── scripts
    └──  ...
```

with the current `json` definition for a connection changing to an array of definitions (either treating the first one as the default or adding a new field `defaultConnection: Boolean`.

When encountering more than one, the IDE should probably ask its user which should be imported (just the default or a subset of the definitions available)

**Comment by adpi2** [comment]
I agree it is useful for a build client like IntelliJ to connect to several build servers but I don't think we need to change BSP for that.
Because BSP already allows to have serveral `.json` file in the same `.bsp` folder:
```
.
├── .bsp
│   └── scala-cli-app1.json
│   └── scala-cli-app2.json
├── app1
│   └──  ...
├── app2
│   └──  ...
└── scripts
    └──  ...
```

It's up to the client then, for instance scala-cli, to manage a number of BSP json file in the same  `.bsp` folder.

**Comment by Gedochao** [comment]
@adpi2 I dunno if the `BSP` docs are up to date, but referring to the [Server Discovery](https://build-server-protocol.github.io/docs/server-discovery.html#the-bsp-connection-details) doc:
```text
BSP connection files:

1. must be unique per build tool name and version to enable different versions of the same build tool to select different BSP connection mechanisms.
2.  (...)
```
If this is true, then then you cannot have multiple `json` definitions for a single build tool under a single `.bsp` folder.

I went ahead and tested this setup in IntelliJ - when I try to import a definition like this in IntelliJ, a random single `BSP` connection is imported. I also see no way to import the others then, since `IntelliJ` seems to expect there's only one connection defined for a directory.

Now, if multiple `json` definitions for a single build tool with different file names are allowed by the protocol, then the docs should be updated to reflect that, and then IDEs should be fixed to support that.

**Comment by jastice** [comment]
It's correct that IntelliJ assumes only one BSP connection file will be used for one `.bsp` directory, so your use case is currently not supported. In principle it would not be a big problem to choose which one to use, internally it's already saved. 

You can have multiple jsons for a single build tool which would run it in different ways. That line maybe needs to be rewritten.

I'm not sure about supporting multiple modules from a single bsp configuration, I think this use case is not common enough to change the specification
